react/features/transcription/components/TranscriptionSubtitles.web.js

```diff
@@ -0,0 +1,329 @@
+     * and the listener will be removed from the conference.
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The read-only props which this Component will
+     * receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (nextProps._conference && !this.state.conference) {
+            const conference = nextProps._conference;
+
+            this.setState({ conference });
+            this.addListener(conference);
+        }
+        if (!nextProps._conference && this.state.conference) {
+            this.removeListener(this.state.conference);
+            this.setState({ conference: undefined });
+        }
+    }
+
+    /**
+     * Add the {@code ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED} listener to
+     * the given conference.
+     *
+     * @param {Object} conference - The conference which has an EventEmitter for
+     * {@code ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED}.
+     * @returns {void}
+     */
+    addListener(conference) {
+        if (conference) {
+            conference.eventEmitter.addListener(
+                ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
+                this.onEndpointMessageReceived.bind(this));
+        }
+    }
+
+    /**
+     * Remove the {@code ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED} listener
+     * from the given conference.
+     *
+     * @param {Object} conference - The conference which has an EventEmitter for
+     * {@code ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED}.
+     * @returns {void}
+     */
+    removeListener(conference) {
+        if (conference) {
+            conference.eventEmitter.removeListener(
+                ConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
+                this.onEndpointMessageReceived);
+        }
+    }
+
+    /**
+     * The CSS to use for rendering the text.
+     */
+    divStyle = {
+        position: 'absolute',
+        bottom: '5%',
+        left: '10%',
+        zIndex: 2,
+        fontWeight: 600,
+        fontSize: '14px',
+        textAlign: 'left',
+        color: '#FFF',
+        opacity: '.80',
+        textShadow: '0px 0px 1px rgba(0,0,0,0.3),'
+                    + '0px 1px 1px rgba(0,0,0,0.3),'
+                    + '1px 0px 1px rgba(0,0,0,0.3),'
+                    + '0px 0px 1px rgba(0,0,0,0.3)'
+    };
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const paragraphs = [];
+
+        this.state.knownIDs.forEach(id => {
+            const o = this.state[id];
+            let text;
+
+            if (o) {
+                text = `${o.name}: `;
+
+                if (o.final) {
+                    text += o.final;
+                } else {
+                    const stable = o.stable ? o.stable : '';
+                    const unstable = o.unstable ? o.unstable : '';
+
+                    text += stable + unstable;
+                }
+            }
```

Comment:
We need to avoid to have too many checks and calculations in the render method. I haven't gone enough into details to understand which exactly are the knownIDs, but if you can find a way to store in the local or the redux state the exact data that needs to be rendered and do those checks at a preliminary stage when we obtain the data it will be better than doing them in the render.